metrics/collectors/shared.lua

```diff
@@ -60,13 +60,19 @@ function Shared:remove(label_pairs)
 end
 
 function Shared:set(num, label_pairs)
+    if num and type(tonumber(num)) ~= 'number' then
+        error("Collector set value should be a number")
+    end
     num = num or 0
     local key = self.make_key(label_pairs)
     self.observations[key] = num
     self.label_pairs[key] = label_pairs or {}
 end
 
 function Shared:inc(num, label_pairs)
+    if num and type(tonumber(num)) ~= 'number' then
```

Comment:
I prefer explicit `num == nil` cause it's more readable, covers `box.NULL` and treats `num == false` case as should.